[visionOS] Get largest image source when using Quick Look for fullscreen

yoelhawa opened the pull request:
#### a59fc6e9e102801c1f9690bb366578dc0a191605
<pre>
[visionOS] Get largest image source when using Quick Look for fullscreen
<a href="https://bugs.webkit.org/show_bug.cgi?id=281628">https://bugs.webkit.org/show_bug.cgi?id=281628</a>
&lt;<a href="https://rdar.apple.com/136328803">rdar://136328803</a>&gt;

Reviewed by Wenson Hsieh.

Upon delegating fullscreen to Quick Look, inform the page that the
viewport size has increased to the maximum possible size. This way,
the largest image source available will be selected from available
`srcset` options.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::didEnterFullScreen):
(WebKit::WebFullScreenManager::didExitFullScreen):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
Inform the page that the viewport size has increased to the maximum
possible size.

Canonical link: <a href="https://commits.webkit.org/285387@main">https://commits.webkit.org/285387@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2fd570a6028cbe22837da8ef98aaf4dc1d881c46

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72449 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51870 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25237 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76630 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23658 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59674 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23480 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/57043 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15561 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75516 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46956 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62392 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/37479 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43607 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22008 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65489 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20216 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78299 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16690 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19349 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65501 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16738 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62404 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64768 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16007 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13039 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6682 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47668 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/2453 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48737 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50026 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48480 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
```diff
@@ -367,6 +367,16 @@ void WebFullScreenManager::didEnterFullScreen()
     }
 
+#if ENABLE(QUICKLOOK_FULLSCREEN)
+    const FloatSize maxViewportSize(10000, 10000);
```
Nit - I think we generally prefer `constexpr` when possible.

```suggestion
    static constexpr auto maxViewportSize = FloatSize { 10000, 10000 };
```

whsieh commented on Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
```diff
@@ -100,6 +100,11 @@ class WebFullScreenManager final : public WebCore::EventListener {
+#if ENABLE(QUICKLOOK_FULLSCREEN)
+    WebCore::FloatSize m_oldSize;
+    double m_scaleFactor;
+    double m_minEffectiveWidth;
```
Let's give these initial values, to avoid potential UB in the future.

```suggestion
    double m_scaleFactor { 1 };
    double m_minEffectiveWidth { 0 };
```

Commit:
[visionOS] Get largest image source when using Quick Look for fullscreen
https://bugs.webkit.org/show_bug.cgi?id=281628
<rdar://136328803>

Reviewed by Wenson Hsieh.

Upon delegating fullscreen to Quick Look, inform the page that the
viewport size has increased to the maximum possible size. This way,
the largest image source available will be selected from available
`srcset` options.

* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::didEnterFullScreen):
(WebKit::WebFullScreenManager::didExitFullScreen):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
Inform the page that the viewport size has increased to the maximum
possible size.

Canonical link: https://commits.webkit.org/285387@main
